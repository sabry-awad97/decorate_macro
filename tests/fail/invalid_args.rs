use decorate_macro::decorate;

fn test_decorator<F, R>(f: F) -> R
where
    F: FnOnce() -> R,
{
    f()
}

#[decorate(test_decorator(invalid,))]
fn test_function(x: i32) -> i32 {
    x + 1
}

fn main() {}
