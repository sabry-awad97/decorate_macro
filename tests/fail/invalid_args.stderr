error: expected expression, found `,`
  --> tests/fail/invalid_args.rs:10:1
   |
10 | #[decorate(test_decorator(invalid,))]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ expected expression
   |
   = note: this error originates in the attribute macro `decorate` (in Nightly builds, run with -Z macro-backtrace for more info)
